src/lua/api-gateway/validation/oauth2/oauthTokenValidator.lua

```diff
@@ -210,14 +210,17 @@ function _M:validateOAuthToken()
             self:setKeyInLocalCache(cacheLookupKey, cachedToken, local_expire_in, "cachedOauthTokens")
             self:setContextProperties(obj)
             return ngx.HTTP_OK
+        elseif (tokenValidity == 0) then
+            ngx.log(ngx.DEBUG, "Token still valid but about to expire in less than 1s")
```

Comment:
Is the token really **valid** then? It seems like the token isn't valid, but the TTL of the cache entry is still valid.

Reply:
Given the fact that we're loosing ms precision with the logic inside `getExpiresIn` function, we might be extending the life of a token with at most 999ms. Changing the message.